### PR TITLE
chore: update dd-trace and remove patch which has been fixed upstream and update nodejs to skip patching

### PR DESCRIPTION
**Pull request by lodmfjord**
# Update dd-trace and remove patch. Update nodejs and generating cache key from node version and yarn version.

TESTING FEW THINGS DO NOT MERGE

A patch was added to fix bug in dd-trace which has been fixed upstream. Also updating since we are many versions  behind and need new datadog features.

Our cache breaks when node is updated (and possibly yarn) - add engines.node and engines.yarn to cache.

Updating node because of changes in dd-trace.

## What

Update dd-trace. Remove patch.

## Why

Less code means less work! We are also having issues with trace in datadog and updating the package (which is few major versions behind) is a good idea. 

## Tests

Running ```test everything``` and ```feature deployment``` to see that this does not break tests and to see how this affects our logging in datadog. 

## Summary by CodeRabbit

- **Chores**
	- Updated Node.js version used in CI/CD pipelines to align with `package.json` settings.
	- Enhanced dependency management in Docker builds by upgrading the Node.js image version.

- **Bug Fixes**
	- Fixed a test issue by introducing a polyfill for `setImmediate` to ensure compatibility and functionality.

- **Documentation**
	- Updated `package.json` to reflect new version changes in dependencies, improving traceability and support.

### CODE REVIEW

File: jest.preset.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ const nxPreset = require('@nx/jest/preset').default
 const { transform, ...nxPresetRest } = nxPreset
 const customResolver = path.join(__dirname, 'jest.resolver.js')
 
+/** Hack: setImmediate is not available in one test */
+global.setImmediate = global.setImmediate || ((fn, ...args) => global.setTimeout(fn, 0, ...args));
+
```
[the remark this answers is not in the document]
Comment (lodmfjord) [reply jest.preset.js]: No this is not needed.

[the remark this answers is not in the document]
Comment (lodmfjord) [reply jest.preset.js]: Can you please approve this?

[the remark this answers is not in the document]
Comment (lodmfjord) [reply jest.preset.js]: this is outdated can you skip this  and longer ask for change?

**Review by saevarma — APPROVED** [review]
🚀

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Pull request closed without merging**